### PR TITLE
[new rule] No global imports

### PR DESCRIPTION
**Pull request by juanpcapurro**
depends on #389 

this is my first time near a syntax tree, so I'm looking forward to your feedback!

I also took the liberty of making it a recommended rule, but if there isn't consensus we could just drop the last commit

as always, see technical details in git history

### CODE REVIEW

File: docs/rules/miscellaneous/comprehensive-interface.md
Comment (juanpcapurro) [reply docs/rules/miscellaneous/comprehensive-interface.md:12]: that's from the documentation update, from #389 , which this PR depends on. 

I wanted to get feedback as early as possible, but for that you and other reviewers should look at the changes introduced in the last two commits. 

It seems that it wasn't obvious and I want to be respectful of your time so I'll mark this PR as a draft until #389 is merged, sorry for the mixup

Comment (dbale-altoros) [reply docs/rules/miscellaneous/comprehensive-interface.md:12]: sorry... I was just going quickly through the changes, I saw this and just asked....  I'm not assigned full time to this project. And I'm also new to it. I did not approve the former pr because of that. Waiting a few days if fvitorio (the former maintainer) can check it as well... if not, I will do it.
Your contributions are very valuable, sorry if we can't keep up

File: lib/rules/best-practises/no-global-import.js
Comment (fvictorio) [inline lib/rules/best-practises/no-global-import.js:33]: It would be better to use the `ImportDirective` visitor callback instead of iterating here. Not a big difference, but it's more idiomatic.

File: test/rules/best-practises/no-global-import.js
Comment (fvictorio) [inline test/rules/best-practises/no-global-import.js:47]: The number of warnings should be asserted here too.

Comment (juanpcapurro) [reply test/rules/best-practises/no-global-import.js:47]: done

File: lib/rules/best-practises/no-global-import.js
Comment (juanpcapurro) [reply lib/rules/best-practises/no-global-import.js]: Initially I had an issue by which the ImportDirective callback wasn't being called for the nodes not defining  `symbolAliases`, which are the interesting ones. But I can't replicate it now :upside_down_face:  so it was probably something else.

pushed again using the ImportDirective callback as suggested.

Comment (dbale-altoros) [reply lib/rules/best-practises/no-global-import.js]: seems to me this can be resolved and pr approved 
do you agree @fvictorio ?

**Review by fvictorio — COMMENTED** [review]
Minor comment, feel free to merge after it's fixed.

**Review by dbale-altoros — APPROVED** [review]

**Pull request merged**
